web3swift/src/Contract/Statically Typed/ABIFunction.swift

```diff
@@ -16,17 +16,16 @@ public protocol ABIFunction {
     var contract: EthereumAddress { get }
     var from: EthereumAddress? { get }
     func encode(to encoder: ABIFunctionEncoder) throws
-    func transaction() throws -> EthereumTransaction
 }
 
 public protocol ABIResponse: ABITupleDecodable {}
 
 extension ABIFunction {
-    public func transaction() throws -> EthereumTransaction {
+    public func transaction(gasPrice: BigUInt? = nil, gasLimit: BigUInt? = nil) throws -> EthereumTransaction {
```

Comment:
y?

Reply:
Needed to override gasPrice and/or gasLimit in `ABIFunction`, i.e. estimating with no refunds

Reply:
Could just pass in a different ABIFunction though/make mutable; might seem more ambiguous down at this level? Approved PR so your call.

Reply:
Needs refactor everywhere + change of API. `ABIFunction` protocol allows for read-only values. We should also differentiate with calls, will change later
👍